src/csync/CMakeLists.txt

```diff
@@ -42,11 +33,49 @@ configure_file(csync_version.h.in ${CMAKE_CURRENT_BINARY_DIR}/csync_version.h)
 
 add_library(csync SHARED ${common_SOURCES} ${csync_SRCS})
 
-target_include_directories(
-  csync
-  PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/std
+target_include_directories(csync
+    PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/std
+    # include root source dir in order to make src/common available to the targets
+    PUBLIC ${PROJECT_SOURCE_DIR}/src
+    # make sure auto-generated headers can be included
+    PUBLIC ${PROJECT_BINARY_DIR}
 )
 
+target_compile_definitions(csync
+    PUBLIC -DQT_NO_CAST_TO_ASCII
+    PUBLIC -DQT_NO_CAST_FROM_ASCII
+    PUBLIC -DQT_NO_URL_CAST_FROM_STRING
+    PUBLIC -DQT_NO_CAST_FROM_BYTEARRAY
+    PUBLIC -DQT_USE_QSTRINGBUILDER
+    PUBLIC -DQT_MESSAGELOGCONTEXT #enable function name and line number in debug output
+    PUBLIC -DQT_DEPRECATED_WARNINGS
+    PUBLIC -DQT_NO_FOREACH
+)
+
+if(MEM_NULL_TESTS)
+    target_compile_definitions(csync PUBLIC -DCSYNC_MEM_NULL_TESTS)
+endif(MEM_NULL_TESTS)
+
+if(TOKEN_AUTH_ONLY)
+    message("Compiling with token authentication")
+    target_compile_definitions(csync PUBLIC -DTOKEN_AUTH_ONLY=1)
```

Comment:
I hate this flag as the feature does not exist any more, this is an imaginary no ui config which probably won't even build.

Reply:
Should we remove it? We could use a `message(FATAL_ERROR ...`).
(In fact, not using `message(STATUS ...)` is a problem already.)

Reply:
Would require removing the ifdef from all the code, keep it for now pls.
I was just mentioning my resent...